### DISCUSSION TITLE
Fetching wrong labels of docker-image

### DISCUSSION THREAD

**Discussion by faenschi** [renovatebot/renovate#25292] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.11

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Fetching the image labels with the tag of the pull request

### What you tried so far.

Hello! 
I would like to ask how the labels of a Docker Image are fetched in renovate? In the renovate log i see, that renovate tries to fetch the labels of the **latest** image, which in my opinion is wrong:

```
DEBUG: getLabels(https://xyz/proxy, latest) (repository=s/renovate-playground)
DEBUG: getManifestResponse(https://xyz/proxy, latest, get) (repository=s/renovate-playground)
```

because in the pull request the version will be updated like that:

```
FROM xyz/proxy:1.0.202310121250
TO
FROM xyz/proxy:1.0.202310190004
```

so shouldn't renovate be looking for the labels in the image with tag **1.0.202310190004** instead of **latest**??

regards angela

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment, marked as answer]
Renovate fetches this information _before_ it knows what the update will be

**Comment by faenschi** [comment]
So how does renovate determine the tag? Sometimes it gets the label from "latest" but i also have an example where it gets them from a tag:

`DEBUG: getLabels(https:/guardix, 4.4.2-rhel8) (repository=s/renovate-playground)`

in the exanple above the tag 4.4.2-rhel8 is used and not latest

**Reply by rarkins** [reply]
I don't have the time to dig into the code, but you're welcome to

**Reply by faenschi** [reply]
yes i did now. ok i see. i think the problem is then, that we have old latest tags in our registry...

```
  const latestTag = tags.includes('latest')
      ? 'latest'
      : findLatestStable(tags) ?? tags[tags.length - 1];
```

thx @rarkins

**Reply by faenschi** [reply]
@rarkins But wouldn't it be better if renovate fetches this information if the updated/pr is created? at this point this is just a best guess which tag it should use isn't it?

**Reply by rarkins** [reply]
What is going _wrong_ though? e.g. the sourceUrl for release notes?

**Reply by faenschi** [reply]
I'm just irritated that there is no link on the package name of the PR:

![image](https://github.com/renovatebot/renovate/assets/11943233/92b6cb16-1ef7-4cf5-b500-aa2598070993)

and i thought this could be, because of the wrong labels (because latest instead of the correct tag is fetched). but please 
correct me if my assumption is wrong

**Reply by rarkins** [reply]
It's not _wrong_ labels, it fetches the sourceUrl from the `latest` label by design. If the package has as _wrong_ latest, then you might get _wrong_ results.
